### PR TITLE
Not a final patch, just a start to semi rewrite.

### PR DESCRIPTION
**Pull request by ScottONeal**
In an effort to revive this project and to push some life into it, I wanted to start a development push. 

I added some tests and changed some things in the base module. It would appear that this module meets a very specific use case, maybe we can start to broaden it, in hopes that there will be some more adoption.

This is not a production ready patch, obviously! :smiley:

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -51,13 +53,16 @@ exports.generateFdf = function(data, fileName) {
   return fdf;
 }
 
-exports.generatePdf = function(data, templatePath,callback) {
+exports.generatePdf = function(data, templatePath, callback) {
   var tempName = temp.path({suffix: '.pdf'});
   var tempNameResult = temp.path({suffix: '.pdf'});
 
-  child = spawn("pdftk", ["app/"+templatePath,"fill_form","-","output",tempName,"flatten"]);
+  child = spawn("pdftk", [__dirname+templatePath,"fill_form","-","output",tempName,"flatten"]);
```
Comment (mnorrish) [inline index.js]: :+1:

Comment (mnorrish) [inline index.js]: Or could the implementing code pass is whatever templatePath they'd like instead of being bound to the `__dirname`?

Comment (ScottONeal) [inline index.js]: That's probably a good idea. Should probably something like:

```
process.cwd() + path.normalize(templatePath)
```

**Comment by blaher** [comment]
:+1:

**Comment by ScottONeal** [comment]
Haven't heard anything from this PR in awhile. If enough people are interested we could create a new repo that can continue in development.

**Comment by dommmel** [comment]
@ScottONeal Sorry for being late to the party :)

Is this PR ready to be merged? I thought it is just a starting point? What plans do you have for this lib? Are you interested in being a maintainer?

**Comment by ScottONeal** [comment]
@dommmel, I never really continued the PR because I didn't get any feedback on the initiation. I can continue development and get it to a place where it is more production ready.

After watching this repo for several months, it does look like there is continual interest in it. So, I don't mind jumping on board to bring it to a place where it is more robust and which may entice more people to use it.

Let me know what route you want to go.

**Comment by blaher** [comment]
@ScottONeal I will note, when I tried your pull request, I was welcomed with a some errors. I uncommented back in the GhostScript portion, and everything worked well after that.

**Comment by ScottONeal** [comment]
Yes, I have rethought the GS integration aspect of it. Ideally, I think the GS part would be based on configuration parameter, because it is an external dependency that the end user will mostly likely not need most of the time (in opinion).

I'll have to brush the dust off this PR and recheck somethings... :smile:

**Comment by ScottONeal** [comment]
Okay, I reenabled GhostScript back into the PR.

`generatePDF()` has been made fully asynchronous. Previously, there were a bunch of sync file system operations that have been made async.

 `generatePDF()` now supports conventional callback arguments of (error, data), previously it did not pass back the error argument

Updated package.json to be version 0.1.0. But, I think we should aim, at some point, to make this module 1.0.0

If you'd like, once the PR is merged, you can make me a contributer and I can setup us up with Travis CI and we can start talking about how to make this module 1.0

`npm test` runs on my macbook, but let me know if you have any problems with it.

:beers:

**Comment by dommmel** [comment]
Thanks @ScottONeal !
I am super tied up at the moment and cannot detailed look before end of next week.
I'll just make you a contributor and you go!

**Comment by ScottONeal** [comment]
Okay, sounds good. :+1:

**Pull request merged**